lib/cli/ui/printer.rb

```diff
@@ -0,0 +1,43 @@
+require 'cli/ui'
+
+module CLI
+  module UI
+    class Printer
+      # Print a message to a stream with common utilities.
+      # Allows overriding the color, encoding, and target stream.
+      # By default, it formats the string using CLI:UI and rescues common stream errors.
+      #
+      # ==== Attributes
+      #
+      # * +msg+ - (required) the string to output. Can be frozen.
+      #
+      # ==== Options
+      #
+      # * +:frame_color+ - Override the frame color. Defaults to nil.
+      # * +:to+ - Target stream, like $stdout or $stderr. Can be anything with a puts method. Defaults to $stdout.
+      # * +:encoding+ - Force the output to be in a certain encoding. Defaults to UTF-8.
+      # * +:format+ - Whether to format the string using CLI::UI.fmt. Defaults to true.
+      # * +:graceful+ - Whether to gracefully rescue common I/O errors. Defaults to true.
```

Comment:
Maybe you could mention that `puts` has a return value, and the error isn't actually resolved, just ignored.